nuxeo-services/nuxeo-platform-actions-core/src/main/java/org/nuxeo/ecm/platform/actions/ELActionContext.java

```diff
@@ -87,8 +88,13 @@ public boolean checkCondition(String expression) throws ELException {
         }
 
         // evaluate expression
-        ValueExpression ve = expressionFactory.createValueExpression(originalContext, expr, Boolean.class);
-        return Boolean.TRUE.equals(ve.getValue(originalContext));
+        ValueExpression ve = expressionFactory.createValueExpression(originalContext, expr, expectedType);
+        return (T) ve.getValue(originalContext);
+    }
+
+    @Override
+    public boolean checkCondition(String expression) throws ELException {
+        return Boolean.TRUE.equals(evalExpression(expression, Boolean.class));
     }
```

Comment:
You mean the old `checkCondition` API? It could even be in the ActionContext interface as well 🤔

Reply:
Sorry the location of my comment was misleading (and the old `checkCondition` API is in the interface, right? :thinking:) and i realize we cannot have a good default implementation, but i meant the new `evalExpression` API here, we should have a default implementation in the abstract implementation `AbstractActionContext` to avoid breaking code compatibility on backport (but on master it may be better to keep thinks like this after all), either raising an exception or returning null.

Reply:
no default implementation on `master` as we don't need it, there is one on backports on the interface itself that throws an exception.